packages/loader/container-loader/src/runtimeCompatManager.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	checkLayerCompatibility,
+	type ICompatibilityDetails,
+	// type IProvideCompatibilityDetails,
+} from "@fluid-internal/client-utils";
+import type { ICriticalContainerError } from "@fluidframework/container-definitions";
+import { UsageError } from "@fluidframework/telemetry-utils/internal";
+
+import { pkgVersion } from "./packageVersion.js";
+
+/**
+ * This class manages the state and validation to ensure that the Loader layer is compatible with the Runtime layer.
+ * @internal
+ */
+export class RuntimeLayerCompatManager implements ICompatibilityDetails {
```

Comment:
i'd rather we remove this class, and keep the data (ICompatibilityDetails) and logic (validateCompatibility) separate, especially as we expect the data to change somewhat reguarly